refiner/app/services/testing.py

```diff
@@ -0,0 +1,215 @@
+            "refine_object": None,
+            "refined_document": None,
+        }
+    )
+    for rc_code, cond in rc_to_condition.items():
+        if cond is None:
+            continue
+        entry = condition_map[cond.id]
+        entry["rc_snomed_codes"].append(rc_code)
+        entry["matching_condition"] = cond
+
+    # STEP 3:
+    # map conditions to configurations
+    conditions = [entry["matching_condition"] for entry in condition_map.values()]
+    condition_to_configuration = await map_conditions_to_configurations(
+        db, conditions, jurisdiction_id
+    )
+    for entry in condition_map.values():
+        cond = entry["matching_condition"]
+        entry["matching_configuration"] = condition_to_configuration.get(cond.id)
+
+    # STEP 4:
+    # for each unique condition with a configuration, process and run refinement
+    # (only output if config found)
+    for entry in condition_map.values():
+        cond = entry["matching_condition"]
+        config = entry["matching_configuration"]
+        if not config:
+            # no config==cannot proceed
+            continue
+
+        payload = ConfigurationPayload(configuration=config, conditions=[cond])
+        processed_config = ProcessedConfiguration.from_payload(payload)
+        entry["refine_object"] = processed_config
+
+        sections_to_include = (
+            [sp.code for sp in config.section_processing if sp.action == "retain"]
+            if config.section_processing
+            else None
+        )
+        refined_eicr_str = refine_eicr(
+            xml_files=XMLFiles(xml_files.eicr, xml_files.rr),
+            processed_condition=None,
+            processed_configuration=processed_config,
+            sections_to_include=sections_to_include,
+        )
+
+        # use the first RR code that mapped to this condition for output
+        # TODO: in the future we might want the object to use a list since technically
+        # there could be more than one `rc_snomed_code` that was **in** the RR,
+        # matched to the condition, and has a configuration. picking the first entry
+        # in an index doesn't feel right but we should wait to see how the trace object
+        # evolves and how the response model evolves
+        rr_code_used = entry["rc_snomed_codes"][0]
+        entry["refined_document"] = RefinedDocument(
+            reportable_condition=ReportableCondition(
+                code=rr_code_used,
+                display_name=cond.display_name,
+            ),
+            refined_eicr=refined_eicr_str,
+        )
+
+    # STEP 5:
+    # build final trace list and output only successful refinements
+    trace = list(condition_map.values())
+
+    # output only successful refinements
+    refined_documents = [
+        entry["refined_document"] for entry in trace if entry["refined_document"]
+    ]
+
+    return {"refined_documents": refined_documents}
+
+
+def extract_reportable_conditions_for_jurisdiction(
```

Comment:
Should this function and the others in this file that aren't `independent_testing` be marked "private"?
```suggestion
def _extract_reportable_conditions_for_jurisdiction(
```

Reply:
this is done now